byzcoin/service.go

```diff
@@ -387,6 +409,61 @@ func (s *Service) CheckAuthorization(req *CheckAuthorization) (resp *CheckAuthor
 	return resp, nil
 }
 
+// DownloadState creates a snapshot of the current state and then returns the
+// instances in small chunks.
+func (s *Service) DownloadState(req *DownloadState) (resp *DownloadStateResponse, err error) {
+	s.updateCollectionLock.Lock()
+	defer s.updateCollectionLock.Unlock()
```

Comment:
Could that mean you can potentially hang the whole cothority if the state is big enough by requesting a download to each node which will prevent them to add a new block as the lock is locked ? Not the protocol but the local DBs won't be updated with the new block until the lock is freed.